### PR TITLE
/tests/overview: Change sorting to show all machine variants together

### PR DESCRIPTION
**Pull request by okurz**
openqa_changed_sorting_with_at_sign

### CODE REVIEW

File: lib/OpenQA/WebAPI/Controller/Test.pm [hunk 1]
```diff
@@ -456,6 +456,7 @@ sub prepare_job_results {
             && $preferred_machines->{$job->ARCH} ne $job->MACHINE)
         {
             $test .= "@" . $job->MACHINE;
+            #$test .= ' (' . $job->MACHINE . ')';
```
Comment (foursixnine) [inline lib/OpenQA/WebAPI/Controller/Test.pm]: Only if this was the line 469... :P

**Comment by okurz** [comment]
Updated. Tests pass, coverage dropped but that unfortunately looks again like unrelated changes about worker registration or something

**Comment by foursixnine** [comment]
@okurz Yes, @dasantiago is looking for possible ways how to cover those lines...

File: templates/test/overview_result_table.html.ep [hunk 2]
```diff
@@ -10,7 +10,8 @@
         </tr>
     </thead>
     <tbody>
-        % for my $config (sort keys %$type_results) {
+        % my @configs = sort { $a =~ s/@/ /r cmp $b =~ s/@/ /r } keys $type_results;
```
Comment (coolo) [inline templates/test/overview_result_table.html.ep]: wasnt't there something about keys on hash references being experimental? 

aka: I believe you are missing the % here.

Comment (coolo) [reply templates/test/overview_result_table.html.ep]: Just as I thought:

coolo@f84#~>perl -w -e 'my $a = { a => 1, b => 2 }; print keys $a;'
Experimental keys on scalar is now forbidden at -e line 1.
Type of arg 1 to keys must be hash or array (not private variable) at -e line 1, near "$a;"

Comment (foursixnine) [reply templates/test/overview_result_table.html.ep]: Wouldn't it make sense to add a perlcritic check for this?

Comment (coolo) [reply templates/test/overview_result_table.html.ep]: it's hard to perlcritic template code unfortunately :(

**Comment by coolo** [comment]
ping

File: templates/test/overview_result_table.html.ep (hunk 2 above)
Comment (okurz) [reply templates/test/overview_result_table.html.ep]: fixed

**Comment by okurz** [comment]
updated:
* rebased
* fixed hash ref dereferencing

**Comment by okurz** [comment]
@coolo

```
coercion for "id" failed: When passing in an object to the WebElement id attribute, it must have at least one of the ELEMENT or element-6066-11e4-a52e-4f735466cecf keys. at /home/travis/perl_modules/lib/perl5/Selenium/Remote/Driver.pm line 968.

 at /home/travis/perl_modules/lib/perl5/Selenium/Remote/Finders.pm line 23.

Can't locate object method "click" via package "0" (perhaps you forgot to load

	"0"?) at t/full-stack.t line 269 (#1)

    (F) You called a method on a class that did not exist, and the method

    could not be found in UNIVERSAL.  This often means that a method

    requires a package that has not been loaded.

    

Uncaught exception from user code:

	Can't locate object method "click" via package "0" (perhaps you forgot to load "0"?) at t/full-stack.t line 269.

# Tests were run but no plan was declared and done_testing() was not seen.
```

seems to be caused by the move to chromedriver?

**Comment by coolo** [comment]
Nope - phantomjs was just as flacky on this test. The error changed though. We're trying to click on an element that is not visible

**Comment by okurz** [comment]
I don't understand. This branch is fully rebased so it should have chromedriver?!? In any case I assume nothing can be done by me here now.

**Comment by coolo** [comment]
yes, you have chromedriver - but the problem is not caused by the switch to it. It was there before.

**Comment by okurz** [comment]
ok, anyway, checks passed now.

**Pull request merged**